docs/cockroachdb.md

```diff
@@ -0,0 +1,391 @@
+  # All statements must be terminated by a semicolon.
+  # To exit, type: \q.
+  #
+  # Server version: CockroachDB CCL v20.2.5 (x86_64-unknown-linux-gnu, built 2021/02/16 12:52:58, go1.13.14) (same version as client)
+  # Cluster ID: e51bfde5-2e75-4991-844e-d769f4b9b684
+  #
+  # Enter \? for a brief introduction.
+  #
+  root@:26257/defaultdb>
+  ```
+- Run some basic SQL queries.
+  Example command:
+  ```bash
+  root@:26257/defaultdb> CREATE DATABASE bank;
+  root@:26257/defaultdb> CREATE TABLE bank.accounts (id INT PRIMARY KEY, balance DECIMAL);
+  root@:26257/defaultdb> INSERT INTO bank.accounts VALUES (1, 1000.50);
+  root@:26257/defaultdb> SELECT * FROM bank.accounts;
+    id | balance
+  -----+----------
+     1 | 1000.50
+  (1 row)
+  ```
+
+- Create database user with password for accessing the database using web UI.
+  ```bash
+  root@:26257/defaultdb> CREATE USER roach WITH PASSWORD 'Q7gc8rEdS';
+  root@:26257/defaultdb> GRANT admin TO roach;
+  ```
+ 
+- Create one more database, which will be used later for running benchmark load
+  ```bash
+  root@:26257/defaultdb> CREATE DATABASE sbtest;
+  root@:26257/defaultdb> \q
+  ```
+ 
+#### Accessing the database's web interface. 
+We will be using NodePort for accessing the service.
+In production environment either use loadbalancer or ingress services as per your requirement
+
+- Create a new node port service using the following
+  ```bash 
+  cat cockroachdb-public-node-port.yaml
+  apiVersion: v1
+  kind: Service
+  metadata:
+    name: cockroachdb-public-nodeport
+    namespace: default
+  spec:
+    ports:
+    - name: grpc
+      port: 26257
+      protocol: TCP
+      targetPort: 26257
+    - name: http
+      port: 8080
+      protocol: TCP
+      targetPort: 8080
+    selector:
+      app.kubernetes.io/component: database
+      app.kubernetes.io/instance: cockroachdb
+      app.kubernetes.io/name: cockroachdb
+    sessionAffinity: None
+    type: NodePort
+  ```
+
+  ```bash
+  kubectl apply -f cockroachdb-public-node-port.yaml
+  ```
+- Get the service details
+  ```bash
+  kubectl get svc
+  ```
+  ```bash
+  #sample output
+  NAME                          TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)                          AGE
+  cockroachdb                   ClusterIP   None          <none>        26257/TCP,8080/TCP               6m57s
+  cockroachdb-public            ClusterIP   10.68.5.179   <none>        26257/TCP,8080/TCP               6m57s
+  cockroachdb-public-nodeport   NodePort    10.68.4.195   <none>        26257:30324/TCP,8080:31937/TCP   5s
+  kubernetes                    ClusterIP   10.68.0.1     <none>        443/TCP                          43m
+  ```
+- Verify that the cockroachDB Dashboard is accessible using web interface
+  ```bash
+  https://<any_node_external-ip>:<NodePort>
+  ```
+  Login credentials for the web UI
+  Username: `roach`
+  Password: `Q7gc8rEdS`
+
+
+      
+### Testing CockroachDB Performance on OpenEBS
```

Comment:
Let's remove the performance tests details. We can keep it under the solution guide only.